plotly/grid_objs/grid_objs.py

```diff
@@ -118,29 +118,96 @@ class Grid(MutableSequence):
     py.plot([trace], filename='graph from grid')
     ```
     """
-    def __init__(self, iterable_of_columns):
+    def __init__(self, columns_or_json, fid=None):
         """
-        Initialize a grid with an iterable of
-        `plotly.grid_objs.Column objects
+        Initialize a grid with an iterable of `plotly.grid_objs.Column`
+        objects or a json/dict describing a grid. See second usage example
+        below for the necessary structure of the dict.
 
-        Usage example:
+        :param (str|bool) fid: should not be accessible to users. Default
+            is 'None' but if a grid is retrieved via `py.get_grid()` then the
+            retrieved grid response will contain the fid which will be
+            necessary to set `self.id` and `self._columns.id` below.
+
+        Example from iterable of columns:
         ```
         column_1 = Column([1, 2, 3], 'time')
         column_2 = Column([4, 2, 5], 'voltage')
         grid = Grid([column_1, column_2])
         ```
+        Example from json grid
+        ```
+        grid_json = {
+            'cols': {
+                'time': {'data': [1, 2, 3], 'order': 0, 'uid': '4cd7fc'},
+                'voltage': {'data': [4, 2, 5], 'order': 1, 'uid': u'2744be'}
+            }
+        }
+        grid = Grid(grid_json)
+        ```
         """
-
         # TODO: verify that columns are actually columns
+        if isinstance(columns_or_json, dict):
+            # check that fid is entered
+            if fid is None:
+                raise exceptions.PlotlyError(
+                    "If you are manually converting a raw json/dict grid "
+                    "into a Grid instance, you must ensure that make "
+                    "'fid' is set to your file ID. This looks like "
+                    "'username:187'."
+                )
+            # TODO: verify that fid is a correct fid if a string
+            self.id = fid
```

Comment:
Actually, if you call `grid.id` you'll get the `fid` back. That was in place before I started working on this PR. so id === fid for the grid.